client/my-sites/email/titan-set-up-mailbox/titan-set-up-mailbox-form.tsx

```diff
@@ -91,10 +86,12 @@ const useHandleCompleteSetup = (
 		try {
 			await mutateAsync( mailbox.getAsFlatObject() as TitanMailboxFields );
 
-			goToThankYouPage(
-				selectedSite?.slug as string,
-				selectedDomainName,
-				mailbox.formFields.mailbox.value
+			// The process is async at teh backend, but typically 5seconds is enough to wait for the mailbox to be created
```

Comment:
```suggestion
			// The provision of new user (mailbox) is done through an async process, which means
			// the new mailbox won't be immediately available. To account for this, we redirect
			// users back to the Email Management page with a 5 seconds delay, to prevent
			// users from seeing the "Set up mailbox" CTA again. There's a chance that the
			// new mailbox is still unavailable after 5 seconds, but that is an edge case that we will
			// cope with for the time being.
```

As discussed in p1708619733142619/1708618647.192489-slack-C06ELHR6L9J, there isn't a bug in the backend. The reason for the mailbox to not show up immediately is because the provision process is done async. Let's update the comment to clarify that!